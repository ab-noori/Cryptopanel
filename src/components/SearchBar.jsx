import React from 'react';
import PropTypes from 'prop-types';
import './SearchBar.scss';

const SearchBar = ({ value, onChange, onKeyDown }) => (
  <div className="search-bar">
    <input
      type="text"
      value={value}
      placeholder="Search"
      onChange={onChange}
      onKeyDown={onKeyDown}
    />
    <button type="button">Search</button>
  </div>
);

SearchBar.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onKeyDown: PropTypes.func.isRequired,
};

export default SearchBar;
