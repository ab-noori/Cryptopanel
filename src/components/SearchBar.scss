.search-bar {
  width: 30%;
  margin: 0.5rem;
  display: flex;
  align-items: center;
  padding: 0.5rem;
  margin-left: 68%;
  background-color: #f2f2f2;
  border-radius: 4px;
}

.search-bar input {
  flex: 1;
  border: none;
  padding: 0.5rem;
  font-size: 1rem;
  background-color: transparent;
}

.search-bar input:focus {
  outline: none;
}

.search-bar button {
  margin-left: 0.5rem;
  padding: 0.5rem;
  border: none;
  border-radius: 4px;
  font-size: 1rem;
  background-color: #333;
  color: #fff;
  cursor: pointer;
}

@media screen and (max-width: 768px) {
  .search-bar {
    width: 90%;
    margin: 0.5rem auto;
  }
}
