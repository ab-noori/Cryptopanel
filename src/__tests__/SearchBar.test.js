import React from 'react';
import { render } from '@testing-library/react';
import SearchBar from '../components/SearchBar';

test('SearchBar component matches snapshot', () => {
  const { container } = render(
    <SearchBar
      value="test value"
      onChange={() => {}}
      onKeyDown={() => {}}
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});
