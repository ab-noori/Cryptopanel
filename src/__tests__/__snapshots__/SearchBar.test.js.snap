// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component matches snapshot 1`] = `
<div
  class="search-bar"
>
  <input
    placeholder="Search"
    type="text"
    value="test value"
  />
  <button
    type="button"
  >
    Search
  </button>
</div>
`;
